static/inline/js/index.js

```diff
@@ -1,50 +1,56 @@
-        main.innerHTML = data.inlineTemplate;
-
-        frag.appendChild(main);
-        target.appendChild(frag);
-    },
-    isset = function(o,p){
-            var val = o;
-            if (p) val = o[p];
-            return 'undefined' !== typeof val;
-        };
-
-    var body = document.body;
-
-    var initialize = function(){       
-        var settings = {};
-
-        if (Utils.isset(parent) && Utils.isset(parent,'__TVPage__') && Utils.isset(parent.__TVPage__, 'config')) {
-            settings = parent.__TVPage__.config[body.getAttribute('data-id')];
+    var utils = {
+        dBody: document.body,
+        render: function(data){
+            if (!this.dBody) return;
+            var frag = document.createDocumentFragment(),
+                main = document.createElement('div'),
+                d = data || {};
+            main.id = d.id || '';
+            main.classList.add('iframe-content');
+            main.innerHTML = (Utils.tmpl(data.inlineTemplate, data.settings.videoData[0])? Utils.tmpl(data.inlineTemplate, data.settings.videoData[0]) : data.inlineTemplate);
+            frag.appendChild(main);
+            this.dBody.appendChild(frag);
         }
+    };
 
-        render(body,{
+    var initialize = function(settings){   
+    if (!settings.videoData || !settings.videoData.length)return;  
+        utils.render({
             id: settings.name,
             title: settings.title || 'Recommended Videos',
-            inlineTemplate: settings.templates.inline
+            inlineTemplate: settings.templates.inline['content'],
+            settings: settings
+        });
+
+        Utils.loadProducts(settings.videoData[0].id, settings.videoData[0].loginId,function(data){
+            settings.productsFirstData = data;
         });
 
-        var checks = 0;
-        (function libsReady(){
+        var libChecks = 0;
+        (function libChecker(){
             setTimeout(function(){
-                if ( (!isset(window,'TVPage') || !isset(window,'_tvpa')) && (++checks < 200) ) {
-                    libsReady();
+                if ( (!Utils.isset(window,'TVPage') || !Utils.isset(window,'_tvpa') || !Utils.isset(window,'Inline') || !settings.productsFirstData || !settings.productsFirstData.length) && (++libChecks < 200) ) {
+                    libChecker();
                 }
                 else{
-                    Inline(settings.name, settings);
+                    var inline = new Inline(settings);
+                    inline.init();
                 }
-            }, 150);
+            }, 100);
         })();
     };
-    
-    initialize();
+
+    var checks = 0;
+    (function isReady(){
+        var settings = parent.__TVPage__.config[utils.dBody.getAttribute('data-id')];
+        setTimeout(function(){
+            if (!Utils.isset(parent) || !Utils.isset(parent,'__TVPage__') || !Utils.isset(parent.__TVPage__, 'config' || !settings.videoData || !settings.videoData.length) && (++checks < 200) ) {
```

Comment:
Looks like we are using `settings.videoData` a few times, can you put it in a variable?